libs/androidlib/src/mill/androidlib/AndroidSdkModule.scala

```diff
@@ -263,6 +261,104 @@ trait AndroidSdkModule extends Module {
     PathRef(ndkPath().path / "build" / "cmake" / "android.toolchain.cmake")
   }
 
+  private def isCI: Boolean = {
+    val ciEnvironments = Seq(
+      "CI",
+      "CONTINUOUS_INTEGRATION",
+      "JENKINS_URL",
+      "TRAVIS",
+      "CIRCLECI",
+      "GITHUB_ACTIONS",
+      "GITLAB_CI",
+      "BITBUCKET_PIPELINE",
+      "TEAMCITY_VERSION"
+    )
+    ciEnvironments.exists(env => sys.env.contains(env))
+  }
+
+  private def cmdlineToolsURL(versionLong: String): String = {
+    val osName = System.getProperty("os.name").toLowerCase
+
+    val platform = Seq("linux", "mac", "windows").find(osName.contains) match {
+      case Some(p) => p
+      case None =>
+        throw new IllegalStateException(s"Unsupported platform for cmdline tools: $osName")
+    }
+
+    s"https://dl.google.com/android/repository/commandlinetools-$platform-${versionLong}_latest.zip"
+  }
+
+  private def installCmdlineTools(
+      sdkPath: os.Path,
+      versionShort: String,
+      remoteReposInfo: os.Path
+  ) = {
+    object CmdlineToolsHelper {
```

Comment:
I think there's no need for this nested object

Reply:
Alternatively , this can be a separate task so "system" or "mill" versions stay separated and also can be independently customised